### DISCUSSION TITLE
Horizontal scalability on k8s

### DISCUSSION THREAD

**Discussion by fabricebrito** [reflex-dev/reflex#3208] [General]
I'm wondering how a reflex app can scale on k8s. I guess that with several users accessing an app, a single pod won't be enough.

Would scaling additional frontend and backend containers in different pods work?

**Comment by masenf** [comment]
The frontend of the app _should_ be exported as static html/css/js files via `reflex export --frontend-only`. This can be served from a loadbalancer or static server, s3 bucket, etc.

The backend of the app can scale to as many instances as required, provided all are using the same `REDIS_URL` to ensure that the state can be accessed regardless of which instance the user comes in on.

Take care when uploading files to store them to some kind of centralized storage like s3, otherwise the builtin `rx.get_upload_dir` will be per-backend and not work across different backend instances.

**Reply by fabricebrito** [reply]
Thanks! 

That makes sense. So a helm chart would include a deployment/service for the backend, a static deployment for the frontend and finally a redis deployment.
